exercises/t_jubb/ex01_basics/src/find_words.cpp

```diff
@@ -0,0 +1,189 @@
+
+    @param vec : a vector to append the entries of count_map to.
+    @param count_map : a map of <string,value> pairs.
+
+  */
+  std::pair<int,std::string> elem;
+  for ( std::map<std::string, int >::iterator i = count_map.begin() ; i != count_map.end(); i++ ) {
+      // Reverse the pair to allow the default vector sort to compare the first entry.
+      elem.first  = (*i).second;
+      elem.second = (*i).first;
+      vec.push_back(elem);
+  };
+  std::sort(vec.begin(),vec.end());
+  return 0;
+}
+
+int main()
+{
+const bool verbose = false;
+
+// Map to store the words and counts and key/value pairs
+std::map<std::string, int> counts;
+
+// Prompt user for a filepath and check if file exists
+std::string file_path;
+cout << "Please enter a valid filepath : " ;
+bool valid_file = false;
+while (!valid_file){
+  std::cin >> file_path;
+  std::ifstream infile(file_path);
+  if (!infile.good() ){
+    cout << "File does not exist, please enter a valid file : ";
+  }
+  else {
+    valid_file = true;
+  }
+}
+
+std::fstream myFile;
+myFile.open(file_path,std::ios_base::in);
+
+if (myFile.is_open()) {
+
+  cout << "File " << file_path << " opened successfully!" << std::endl;
+  std::string file_line;
+  int line = 0;
+
+  while (myFile.good()) {
+    // calculate line by line
+    line++;
+    std::getline(myFile,file_line);
+    if (verbose == true) {
+      cout << "\tFile Line " << line << " : " << file_line << endl;
+    }
+
+    // find/count the words and add to counts map
+    std::vector<std::string> words;
+    words = get_words(file_line, 4);
+    if (verbose == true) {
+      int counter = 0;
+      for ( auto i = words.begin(); i != words.end(); i++ ) {
+          counter++;
+          std::cout << counter << " : " << *i << std::endl;
+      }
+    }
+    add_word_counts(counts , words);
+  }
+  myFile.close();
+}
+
+// Display the words and counts
+cout << endl << "Word Occurance : " << endl << endl;
+unsigned int counter = 0;
+for ( std::map<std::string, int>::iterator i = counts.begin(); i != counts.end(); i++ ) {
+    counter++;
+
+    // Pad the gap between word and count.
+    int pad = 15 - (*i).first.size();
+    std::string pad_blank = " " ;
+    for (int i=1; i<pad; i++) {
+      pad_blank = pad_blank + " ";
+    }
+
+    // Pad the index of the words
+    int num_pad = 3; 
+
+    cout << std::setfill('0') << std::setw(num_pad) << counter << " : ";
+    cout << (*i).first << pad_blank << (*i).second << endl;
+}
+
+// Order the pairs by value using default vector sort
+std::vector< std::pair<  int , std::string > > ordered;
+ordered_by_value(ordered, counts);
+
+// Display the ordered words and counts.
+cout << endl << "Ordered By Occurance : " << endl << endl;
+counter = 0;
+for ( std::vector< std::pair<  int , std::string > >::iterator i = ordered.begin(); i != ordered.end(); i++ ) {
+    counter++;
+
+    // Pad the gap between word and count.
+    int pad = 15 - (*i).second.size();
+    std::string pad_blank = " " ;
```

Comment:
This could be done without the for loop using one of the string constructors (http://en.cppreference.com/w/cpp/string/basic_string):
```
std::string pad_blank(pad, ' ');
```